sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AggregateFunctions.scala

```diff
@@ -418,6 +418,63 @@ class CudfLastExcludeNulls(ref: Expression) extends CudfFirstLastBase(ref) {
   override val offset: Int = -1
 }
 
+/**
+ * This class is only used by the M2 class aggregates, do not confuse this with GpuAverage.
+ * In the future, this aggregate class should be removed and the mean values should be
+ * generated in the output of libcudf's M2 aggregate.
+ */
+class CudfMean(ref: Expression) extends CudfAggregate(ref) {
+  @transient val rapidsAvgType: DType = GpuColumnVector.getNonNestedRapidsType(dataType)
```

Comment:
seems like this one is not used not sure if I had it there before, but please check if it can be removed.

Reply:
It is used: https://github.com/NVIDIA/spark-rapids/pull/3373/files#diff-7f3472a1577f44fa821a6156d356a32dcb0bdd4edd0873570dba59c31601528cR1150

Reply:
No I meant `rapidsAvgType`, I don't see that referenced anywhere.